File: app/src/main/java/com/zxy/demo/HttpInterceptor.java
package com.zxy.demo;


import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;


/**
 * Created by zxy on 2019/08/06.
 */
public class HttpInterceptor implements Interceptor {
    @NotNull
    @Override
    public Response intercept(@NotNull Chain chain) throws IOException {
        Request request = chain.request();
        request = request.newBuilder()
                .addHeader("platform","android")
                .addHeader("channel","2048_android")
                .addHeader("version","1.1.0")
                .addHeader("Authorization","Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOlwvXC9hcGkudGVzdC4yMDQ4LmNvbVwvYXBpXC9yZWZyZXNoIiwiaWF0IjoxNTczMDE4NzE5LCJleHAiOjE1NzM2MjY3ODcsIm5iZiI6MTU3MzAyMTk4NywianRpIjoiSThrdEZIWGJlWEQxaUhsZyIsInN1YiI6MjQ0LCJwcnYiOiIyM2JkNWM4OTQ5ZjYwMGFkYjM5ZTcwMWM0MDA4NzJkYjdhNTk3NmY3In0.GgTpWFg1ay8RJD13Dj55UuoSjj8C_pdza27Fs0ZFVnI")
                .addHeader("system","XiaoMi MIX2:0")
                .addHeader("Accept","application/x.2048.v2+json")
                .build();
        Response response = chain.proceed(request);
        return response;
    }
}

File: app/src/main/java/com/zxy/demo/HttpInterceptor.kt
package com.zxy.demo

import android.content.Context
import com.zxy.demo.ServiceInterface.Companion.token
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException
import java.util.*

/**
 * Created by Admin
 * on 2020/11/13
 */
class HttpInterceptor : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        var request = chain.request()
        if ("POST" == request.method) {
            request = request.newBuilder()
                .addHeader("Authorization", "Bearer $token")
                .addHeader("oks-deviceid", "bf661c9459f47d09")
                .addHeader("oks-platform","1")
                .addHeader("oks-channel", "oks-channel")
                .addHeader("oks-version", "1.4.0")
                .addHeader("oks-versioncode", "10")
                .addHeader("oks-timezone", TimeZone.getDefault().id)
                .addHeader("oks-locale", "en")
                .addHeader("oks-country-code", "TH")
                .build()
        }
        return chain.proceed(request)
    }
}